airbyte-webapp/STYLE-GUIDE.md

```diff
@@ -0,0 +1,56 @@
+# Frontend Style Guide
+
+This serves as a living document regarding conventions we have agreed upon as a frontend team. In general, the aim of these decisions and discussions is to both (a) increase the readability and consistency of our code and (b) decrease day to day decision-making so we can spend more time writing better code.  
+
+## General Code Style and Formatting
+
+* Where possible, we rely on automated systems to maintain consistency in code style 
+* We use eslint, Prettier, and VSCode settings to automate these choices.  The configuration files for these are checked into our repository, so no individual setup should be required beyond ensuring the [eslint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) and [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) extensions are installed in VSCode
+* Don’t use single-character names. Using meaningful name for function parameters is a way of making the code self-documented and we always should do it. Example:
+    * .filter(([key, value]) => isDefined(value.default) ✅
+    * .filter(([k, v]) => isDefined(v.default) ❌
+
+
+## Exporting
+
+* Export at declaration, not at the bottom. For example:
+    * export const myVar ✅
+    * const myVar; export { myVar }; ❌
+
+
+## File Naming
+
+* TODO
+
+
+## Git
+
+* All branches should be &lt;my name>/feature-or-bug-description, for example:
+    * kc/generated-api-client
+    * teal/deleted-connections-ui
+    * edmundo/fix-pwd-input-vis-btn
+
```

Comment:
Since this is not really related to code-style (and doesn't apply to community contributors), I wonder if we should move this instead separate to the Notion page, maybe under the "Frontend Development Setup" section?